executor/aggfuncs/func_group_concat_test.go

```diff
@@ -45,3 +52,272 @@ func (s *testSuite) TestGroupConcat(c *C) {
+	memDeltas = make([]int64, 0)
+	buffer := new(bytes.Buffer)
+	for i := 0; i < srcChk.NumRows(); i++ {
+		row := srcChk.GetRow(i)
+		if row.IsNull(0) {
+			memDeltas = append(memDeltas, int64(0))
+			continue
+		}
+		oldMemSize := buffer.Cap()
+		if i != 0 {
+			buffer.WriteString(" ")
+		}
+		curVal := row.GetString(0)
+		buffer.WriteString(curVal)
+		memDelta := int64(buffer.Cap() - oldMemSize)
+		memDeltas = append(memDeltas, memDelta)
+	}
+	return memDeltas, nil
+}
+
+func groupConcatOrderUpdateMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType) (memDeltas []int64, err error) {
+	memDeltas = make([]int64, 0)
+	buffer := new(bytes.Buffer)
+	dataGen := getDataGenFunc(dataType)
+	for i := 0; i < srcChk.NumRows(); i++ {
+		row := srcChk.GetRow(i)
+		if row.IsNull(0) {
+			memDeltas = append(memDeltas, int64(0))
+			continue
+		}
+		oldMemSize := buffer.Len()
+		if i != 0 {
+			buffer.WriteString(" ")
+		}
+		curVal := row.GetString(0)
+		buffer.WriteString(curVal)
+		dt := dataGen(i)
+		datumMem := aggfuncs.GetDatumMemSize(&dt)
+		memDelta := datumMem + int64(buffer.Len()-oldMemSize)
+		memDeltas = append(memDeltas, memDelta)
+	}
+	return memDeltas, nil
+}
+
+func groupConcatDistinctUpdateMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType) (memDeltas []int64, err error) {
+	valSet := set.NewStringSet()
+	buffer := new(bytes.Buffer)
+	var encodeBytesBuffer []byte
+	for i := 0; i < srcChk.NumRows(); i++ {
+		row := srcChk.GetRow(i)
+		if row.IsNull(0) {
+			memDeltas = append(memDeltas, int64(0))
+			continue
+		}
+		encodeBytesBuffer = encodeBytesBuffer[:0]
+		curVal := row.GetString(0)
+		encodeBytesBuffer = codec.EncodeBytes(encodeBytesBuffer, hack.Slice(curVal))
+		joinedVal := string(encodeBytesBuffer)
+		if valSet.Exist(joinedVal) {
+			memDeltas = append(memDeltas, int64(0))
+			continue
+		}
+		oldMemSize := buffer.Cap()
+		if i != 0 {
+			buffer.WriteString(" ")
+		}
+		valSet.Insert(joinedVal)
+		buffer.WriteString(curVal)
+		memDelta := int64(len(joinedVal) + (buffer.Cap() - oldMemSize))
+		memDeltas = append(memDeltas, memDelta)
+	}
+	return memDeltas, nil
+}
+
+func groupConcatDistinctOrderUpdateMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType) (memDeltas []int64, err error) {
+	valSet := set.NewStringSet()
+	buffer := new(bytes.Buffer)
+	var encodeBytesBuffer []byte
+	dataGen := getDataGenFunc(dataType)
+	for i := 0; i < srcChk.NumRows(); i++ {
+		row := srcChk.GetRow(i)
+		if row.IsNull(0) {
+			memDeltas = append(memDeltas, int64(0))
+			continue
+		}
+		encodeBytesBuffer = encodeBytesBuffer[:0]
+		curVal := row.GetString(0)
+		encodeBytesBuffer = codec.EncodeBytes(encodeBytesBuffer, hack.Slice(curVal))
+		joinedVal := string(encodeBytesBuffer)
+		if valSet.Exist(joinedVal) {
+			memDeltas = append(memDeltas, int64(0))
+			continue
+		}
+		oldMemSize := buffer.Len()
+		if i != 0 {
+			buffer.WriteString(" ")
+		}
```

Comment:
should we use a new const here instead of " " here and testMultiArgsAggFunc? IMO, It will be easier to understand.